server/src/services/telemetry.service.ts

```diff
@@ -1,14 +1,62 @@
 import { snakeCase } from 'lodash';
 import { OnEvent } from 'src/decorators';
-import { ImmichWorker, JobStatus } from 'src/enum';
+import { ImmichTelemetry, ImmichWorker, JobStatus, QueueName } from 'src/enum';
 import { ArgOf, ArgsOf } from 'src/repositories/event.repository';
 import { BaseService } from 'src/services/base.service';
 
+const QUEUE_METRICS_POLLING_INTERVAL = 5000;
+
 export class TelemetryService extends BaseService {
+  private queueCounts = new Map<string, number>();
+  private pollingInterval?: NodeJS.Timeout;
+
   @OnEvent({ name: 'AppBootstrap', workers: [ImmichWorker.Api] })
   async onBootstrap(): Promise<void> {
     const userCount = await this.userRepository.getCount();
     this.telemetryRepository.api.addToGauge('immich.users.total', userCount);
+
+    const { telemetry } = this.configRepository.getEnv();
+    if (telemetry.metrics.has(ImmichTelemetry.Job)) {
+      // Register observable gauges for queued metrics
+      this.registerQueuedMetrics();
+
+      // Start polling queue statistics
+      await this.updateQueuedMetrics();
+      this.pollingInterval = setInterval(() => {
+        void this.updateQueuedMetrics();
+      }, QUEUE_METRICS_POLLING_INTERVAL);
+    }
+  }
+
+  @OnEvent({ name: 'AppShutdown' })
+  onShutdown(): void {
+    if (this.pollingInterval) {
+      clearInterval(this.pollingInterval);
+    }
+  }
+
+  private registerQueuedMetrics(): void {
+    for (const queueName of Object.values(QueueName)) {
+      const metricName = `immich.queues.${snakeCase(queueName)}.queued`;
+      this.telemetryRepository.jobs.setObservableGauge(metricName, () => this.queueCounts.get(metricName) ?? 0, {
+        description: `Number of queued jobs in ${queueName} queue (waiting + paused + delayed)`,
+      });
+    }
+  }
+
+  private async updateQueuedMetrics(): Promise<void> {
+    await Promise.all(
+      Object.values(QueueName).map(async (queueName) => {
+        try {
+          const stats = await this.jobRepository.getJobCounts(queueName);
+          const queuedCount = stats.waiting + stats.paused + stats.delayed;
```

Comment:
Could the different counters be tracked separately? I'd find it useful to alert on an increase in queue delays, if this is wrapped up with the figures from waiting and paused it can't be alerted in the same way. Similarly with waiting and paused, if the total number is taking a long time to come down I wont be able to see if its caused by my server being slow to get through waiting items or if they're in a paused state.